### PR TITLE
[SEMI-MODULAR] [READY] Adds a variant of the PKC available to Ashwalkers.

### PR DESCRIPTION
**Pull request by gonenoculer5**
## About The Pull Request
Adds a new weapon to the Ashwalkers mid-late game arsenal, allowing them access to trophies after a bit of tedium that also requires them to make use of an existing modular system: forging. Do note this is still missing spirtes and should not be merged yet.

Recipe: 
image

To summarize, this version of the PKC is intended for focusing on the core gameplay loop of Ashwalkers whilst also allowing them access to trophies should they wish to take on mega fauna and want a more tribal friendly equivilent. This PKC is much faster and more efficient at butchery (allowing more products per harvest by about 10%, and an arguably higher chance for getting trophies from butchering), at the cost of a higher charge up time, lowered backstab and normal damage. (-5 force, -5 backstab bonus +5 recharge time). Its primary purpose is being a very very effective hunting tool, whilst still being effective enough to fight megafauna, albiet at a disadvantage compared to a standard PKC.

Butchery:
image

## How This Contributes To The Skyrat Roleplay Experience
Primarily this encourages people to learn forging a rather enjoyable system we've added that makes up a secondary gameplay loop for the Ashwalkers, producing armor and weapons with the reagent forge and gathering lavaland materials to unlock reagent enchanting for stronger tools, made from grown reagents or ones traded from miners. The recipe is balanced with the intention that ashwalkers could in theory get this in the earlygame if they were swift, but at the risk of dying earlier in the shift, if there are no other ashwalkers around, whilst opening up content that already exists to another group that could use easier access to said content, considering we already see a fair number of ashwalkers going about with often stolen PKC's or traded for ones as it is.

## Changelog

:cl: Gonenoculer5
add: Ashwalkers have discovered how to construct a makeshift equivilent to the PKC, known as the Proto-Kinetic Axe. Check the crafting menu for the recipe.
/:cl:

### CODE REVIEW

**Comment by gonenoculer5** [comment]
A reminder to _please_ not merge this until the sprites are done.

**Comment by gonenoculer5** [comment]
Also maintainers, if any balance changes need made to this, please let me know. I've very thoroughly tested it on private and everything works as intended, whilst still keeping the enjoyable rhythm of the PKC.

**Comment by gonenoculer5** [comment]
Also. Alternative name suggestions are appreciated.

**Comment by Stalkeros2** [comment]
I don't really think that giving ashies mining tools, even reflavored, would be a good idea... in *any* case.

**Comment by gonenoculer5** [comment]
> I don't really think that giving ashies mining tools, even reflavored, would be a good idea... in _any_ case.

I mean, with the amount of effort required to get your hands on one, I would say its pretty reasonable. Ive spoken to several ashwalker players who think this is a nice change, which is why I PRd it.

**Comment by tf-4** [comment]
Ehhh I kinda like the idea of them trading for stuff from the station. If we just give them their own version of stuff then it removes that kind of beneficial interaction, even if small.

**Comment by gonenoculer5** [comment]
> Ehhh I kinda like the idea of them trading for stuff from the station. If we just give them their own version of stuff then it removes that kind of beneficial interaction, even if small.

They can still trade with the station for PKAs to make them. Ive seen people trade comodities for mining tools before.

**Comment by gonenoculer5** [comment]
And technically it doesnt prevent that. The PKC is still king where it once was. The new PKC has a tradeoff specifically for ashies who either want the challenge or cant trade for one. Its more hunting and gathering focused, hence the butchery speed and efficiency increase, but lower overall damage.

**Comment by Elli-Skala** [comment]
So a few points I'd like to give

The bone axe is in abit of a bad place right now and not really used at all, so now it's just became a component really, so a idea would be just to push the PKC functionality down to the bone axe itself instead of a new weapon 

But also as a side note the increased inefficiency goes against the design of ash walkers suppose to drag their kills back to the nest.

**Comment by gonenoculer5** [comment]
> So a few points I'd like to give
> 
> The bone axe is in abit of a bad place right now and not really used at all, so now it's just became a component really, so a idea would be just to push the PKC functionality down to the bone axe itself instead of a new weapon
> 
> But also as a side note the increased inefficiency goes against the design of ash walkers suppose to drag their kills back to the nest.

Finally someone actually gives constructive criticism other than "ashwalkers shouldn't have this", thank you. (refering to the feedback from the discord, not here.) I'll agree the bone axe is in a bit of a rough spot, I intend to address that at some point in the near future, its definitely on my radar. Have to give it something to make it unique enough to use other than high damage, otherwise the spear wins 9/10 times for equipment choices. 

As for your second point, ashwalkers arent always able/required to drag corpses back. Yes their primary point is to make eggs to expand the nest, but once you have 10-20 or so eggs (which you should by midgame, where you are intended to make this.) then you are free to butcher them as needed for the extra materials. Often even early game if we have a decent number of eggs or its lowpop, I will go as far as to take a mining cart out and collect resources from tendrils before heading home with alot of materials in tow, as there is no real economical way to transport more than two simplemob bodies at a time atm.

**Comment by gonenoculer5** [comment]
As for pushing the PKC functionality down to the bone axe itself? I considered it. I still am partially. But the main reason I am against that is because of the fact that the axe does serve a purpose currently, even a small one, and I dont want assistants running around with a reasonably high force weapon, hence why this is locked to ashies and the bone axe isnt. Plus modularity is preferable in my opinion to modifying existing TG code.

**Comment by dawsonkeyes** [comment]
Rather than not giving this to ash walkers, it needs no repeated use ranged functionality. I would suggest removing ranged and mining capability and making a mark detonation happen after two melee strikes.

Though really, their current gameplay loop is fine. This is a needless buff and they really shouldn’t have it anyways.

**Comment by gonenoculer5** [comment]
> Rather than not giving this to ash walkers, it needs no repeated use ranged functionality. I would suggest removing ranged and mining capability and making a mark detonation happen after two melee strikes.
> 
> Though really, their current gameplay loop is fine. This is a needless buff and they really shouldn’t have it anyways.

Mark detonation part isnt gonna happen, thatd be a pain to code. The charge time had been adjusted to make it impractical for mining. Sure you could use it in a pinch to get away if theres only one or two walls between you and being killed by a goliath, but you arent gonna be running from a colly with it.

**Comment by dawsonkeyes** [comment]
> > Rather than not giving this to ash walkers, it needs no repeated use ranged functionality. I would suggest removing ranged and mining capability and making a mark detonation happen after two melee strikes.
> > Though really, their current gameplay loop is fine. This is a needless buff and they really shouldn’t have it anyways.
> 
> Mark detonation part isnt gonna happen, thatd be a pain to code. The charge time had been adjusted to make it impractical for mining. Sure you could use it in a pinch to get away if theres only one or two walls between you and being killed by a goliath, but you arent gonna be running from a colly with it.

It needs to be a direct downgrade to the pickaxe for mining time, simple as.

**Comment by gonenoculer5** [comment]
> > > Rather than not giving this to ash walkers, it needs no repeated use ranged functionality. I would suggest removing ranged and mining capability and making a mark detonation happen after two melee strikes.
> > > Though really, their current gameplay loop is fine. This is a needless buff and they really shouldn’t have it anyways.
> > 
> > 
> > Mark detonation part isnt gonna happen, thatd be a pain to code. The charge time had been adjusted to make it impractical for mining. Sure you could use it in a pinch to get away if theres only one or two walls between you and being killed by a goliath, but you arent gonna be running from a colly with it.
> 
> It needs to be a direct downgrade to the pickaxe for mining time, simple as.

Cant do that without ruining the rhythm of how the PKC operates. I could bump it up a bit more but in my testing it was fine.

**Comment by gonenoculer5** [comment]
Added lefthand and righthand sprites for the axe. Need to fix the alignment, its like 10 pixels higher than it should be.

**Comment by gonenoculer5** [comment]
> Added lefthand and righthand sprites for the axe. Need to fix the alignment, its like 10 pixels higher than it should be.

image

**Comment by PCPort** [comment]
As a miner main, I endorse this PR.

**Comment by gonenoculer5** [comment]
https://gyazo.com/f46f7ce000520a0fc2ccd322c664b525

**Comment by gonenoculer5** [comment]
@GoldenAlpharex Sprites are done, DNM tag can now be removed and this is ready to be merged.

**Comment by gonenoculer5** [comment]
Actually one moment I need to make sure the ashies can actually craft it and station crew cant.

**Comment by gonenoculer5** [comment]
Alright we're good to go now. Merge when ready.

**Comment by GoldenAlpharex** [comment]
I haven't taken the time to review this just yet, but personally I'm not sure that I'm too in favor of this, for multiple reasons:
1. Ash Walkers are meant to be a clear, primitive downgrade to miners. While this isn't exactly a sidegrade to the PKC for reasons mentioned before, I still feel like it gives them yet another tool to try and take from the miners' job. Giving them the ability to create a unique, stronger weapon kind of defeats the purpose of them not being meant to replace miners.
2. Trophies. I feel like, if this has no kinetic detonation (read: the blue projectile that causes the weird blue sphere on mobs on hit), many of the trophies will kind of feel useless. Furthermore, I don't want them to get said kinetic detonation, so adding it to justify "Well we can already use trophies but they don't do anything in most cases" is not something I want to go for.
3. Butchering. I feel like giving a bonus to butchering goes directly against the spirit of dragging the corpses back to the tendril to properly harvest them. It even gives them a *bonus* at doing that rather than bringing stuff back at the tendril.

Also, now that I've gone to take a peak at the code just a bit to see if there was any mentions of the effects of the trophies on it, I don't really see any major mentions, so I'd like more explanations as to how they're meant to interract with those.

File: modular_skyrat/modules/bonecrusher/bone_crusher.dm
Comment (GoldenAlpharex) [inline modular_skyrat/modules/bonecrusher/bone_crusher.dm:7]: This feels like a lot of comment for stuff that probably should've been in the PR body.

File: modular_skyrat/modules/bonecrusher/bone_crusher.dm
Comment (GoldenAlpharex) [inline modular_skyrat/modules/bonecrusher/bone_crusher.dm:49]: I cannot be approving code with blatant copy-paste when the value of `force_wielded` could be a variable stored on the crusher itself.

You can do much better than this, and it would save a lot of copy-paste, and would make the overall code much cleaner.

File: modular_skyrat/modules/bonecrusher/bone_crusher.dm
Comment (GoldenAlpharex) [inline modular_skyrat/modules/bonecrusher/bone_crusher.dm:186]: How are they meant to access a PKA?

File: modular_skyrat/modules/bonecrusher/bone_crusher.dm
Comment (GoldenAlpharex) [inline modular_skyrat/modules/bonecrusher/bone_crusher.dm:177]: I'm not going to read all of this because it's probably mostly copy-paste, so please make this a subtype of the PKC if you want this to go through.

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]
This really should be made a child of the normal PKC. However, fixing that doesn't mean that I would approve of this, I still would like to be convinced about the three points I brought up in the above comment.

File: modular_skyrat/modules/bonecrusher/bone_crusher.dm
Comment (gonenoculer5) [reply modular_skyrat/modules/bonecrusher/bone_crusher.dm:49]: Half the reason it was fully copy pasted instead of made a subtype was for the freedom of being able to make more intensive changes to the related code without having to add a bunch of variables, and without directly modifying PKC code itself. This is because of the crushers ComponentInitialize() in the parent. Im not sure how nicely this would play codewise if I changed out force_wielded = 15 to force_wielded = [var name here] and set the var above it in the item for both the PKC and the Bone Crusher.

File: modular_skyrat/modules/bonecrusher/bone_crusher.dm
Comment (gonenoculer5) [reply modular_skyrat/modules/bonecrusher/bone_crusher.dm:186]: Legion corpses rarely drop them, killing the BDM drops one or two, trading with mining.. etc.

**Comment by gonenoculer5** [comment]
> 1. Ash Walkers are meant to be a clear, primitive downgrade to miners. While this isn't exactly a sidegrade to the PKC for reasons mentioned before, I still feel like it gives them yet another tool to try and take from the miners' job. Giving them the ability to create a unique, stronger weapon kind of defeats the purpose of them not being meant to replace miners.

As discussed earlier in the PR comment body, this is absolutely still a downgrade to the PKC. It has an increased charge time to make it impractical for mining, and technically I _could_ try to remove the mining component from the actual projectile but that might cause issues. This will not be ashwalkers "replacing" miners, they arent exactly going to be going out, collecting resources and depositing them. Ashwalkers have very little incentive to actually mine aside from getting iron for the forge or plasma. At worst the mining function of this might be useful as a haphazard escape tool considering a diamond pick is faster than the bone crusher recharge rate. If need be this can be adjusted higher to deincentivize it further but I would not recommend that.


> 2. Trophies. I feel like, if this has no kinetic detonation (read: the blue projectile that causes the weird blue sphere on mobs on hit), many of the trophies will kind of feel useless. Furthermore, I don't want them to get said kinetic detonation, so adding it to justify "Well we can already use trophies but they don't do anything in most cases" is not something I want to go for.

At no point in this PR did I state that trophies would not function as they do normally for the bone crusher. The explicit purpose was to allow Ashwalkers to access the abilities of the PKC whilst not being a overwhelming upgrade or sidegrade to it that would blindside anything else they can make. This weapon is still comprable to a bone spear without any trophies. 

> 3. Butchering. I feel like giving a bonus to butchering goes directly against the spirit of dragging the corpses back to the tendril to properly harvest them. It even gives them a bonus at doing that rather than bringing stuff back at the tendril.

If you've played ashies or read the comment body of this PR, you'd see this problem already addressed. After the ashwalkers reach mid game, generally speaking with a populace of 2-3 ashwalkers at a bare minimum, you already have atleast 10-15 eggs, usually more. After 20 or so eggs it is redundant to continue to sacrifice to the tendril, lest you need to regain a blessing, as such most hunters will opt to butcher in the field and bring the products back to the nest _anyway_, using a knife since it has a higher bonus value than the spear might. This PKC, as stated, is intended for the _mid to late game_, hence why it has a rather signifigant resource sink to create. It requires knowing reagent forging, you either have to have found/stolen or bartered/traded for a PKA, and then ontop of both of these you have to have a rather reasonable amount of materials ontop of the already expensive bone axe in this recipe.

File: modular_skyrat/modules/bonecrusher/bone_crusher.dm
Comment (gonenoculer5) [reply modular_skyrat/modules/bonecrusher/bone_crusher.dm:7]: Temporary comments minus the credit and line edits.

**Comment by gonenoculer5** [comment]
And in summary of the requested review, I _could_ try to make this a subtype as I originally had in mind, but the issue was trying to get force_wielded to be 15 instead of 20. I could try a variable in place of the value, that might work but would require making an edit to PKC code which I would really like to avoid if possible.

**Comment by GoldenAlpharex** [comment]
It took a while for that slow brain of mine to realize that it kept the kinetic blast part, but now I have.

Knowing that, I'm even less into this PR, I really dislike the idea of Ash Walkers having basically a reason to go out and fight megafauna, because those are the things that give the good PKC trophies, especially like, Hierophant, at least.

**Comment by gonenoculer5** [comment]
> It took a while for that slow brain of mine to realize that it kept the kinetic blast part, but now I have.
> 
> Knowing that, I'm even less into this PR, I really dislike the idea of Ash Walkers having basically a reason to go out and fight megafauna, because those are the things that give the good PKC trophies, especially like, Hierophant, at least.

Also. I forgot to mention this but oldbase had something similar to this except it was dropped by a mob rather than crafted.

..Also the Ashwalkers already HAVE reasons to fight fauna. A drake corpse grants ten eggs from the tendril. They are incentivized because all their arrows have a damage bonus to megafauna. Why WOULDN'T they go out and fight them?

For reference, 10 brass arrows is enough to kill a collossus. And thats not at all hard to pull off.

**Comment by MimicFaux** [comment]
In addition, they wouldn't be the only lavaland faction that already does fight megas. Golem miners and Interdyne miners already test their mettle against the bigger challenges of the megas. Ashwalkers, prior to their equipment options getting expanded to include specifically tuned megafauna items like the brass arrows, are incentivized like Gone mentioned with the bonus stockpile to eggs with drakes. As a side note, Ashwalkers have always been the scavengers of mining. Having had crates of hoarded supplies that (presumably) fell from orbiting stations. It wasn't too long ago they had singularity generators in their stockpile. cobbled together PKC is not that far off theme.

File: modular_skyrat/modules/bonecrusher/bone_crusher.dm
Comment (gonenoculer5) [reply modular_skyrat/modules/bonecrusher/bone_crusher.dm:177]: Christ could you not quote half the file. My scroll wheel is screaming here.

**Comment by jjpark-kb** [comment]
Ashwalkers are supposed to be tribal, and by having them work towards this, this seems to be heading away from tribal.

**Pull request closed without merging**
